x-pack/plugins/index_management/public/application/components/mappings_editor/lib/utils.ts

```diff
@@ -602,3 +603,93 @@ export const deNormalizeRuntimeFields = (fields: NormalizedRuntimeFields): Runti
     };
   }, {} as RuntimeFields);
 };
+
+/**
+ * get all the fields from given state which matches selected DataTypes from filter
+ *
+ * @param state The state that we are using depending on the context (when adding new fields, static state is used)
+ * @param filteredDataTypes data types array from which fields are filtered from given state
+ */
+
+export const getFieldsMatchingFilterFromState = (
```

Comment:
We should be able to still simplify this function quite a lot. We're iterating repeatedly over the `state.fields.byId` when we should be able to do it in one pass:

```
  return Object.fromEntries(Object.entries(state.fields.byId).filter(([_, value]) => filteredDataTypes.includes(TYPE_DEFINITION[state.fields.byId[fieldId].source.type].label));
```

Reply:
updated in b9486659f6d6b9380fe02a8a5d04004cf8002c72